src/RoadRegistry.BackOffice.Handlers/Extracts/DownloadFileContentRequestHandler.cs

```diff
@@ -6,39 +6,65 @@ namespace RoadRegistry.BackOffice.Handlers.Extracts;
 using BackOffice.Extracts;
 using Be.Vlaanderen.Basisregisters.BlobStore;
 using Editor.Schema;
+using Editor.Schema.Extensions;
 using Extensions;
 using FluentValidation;
 using FluentValidation.Results;
 using Framework;
 using Messages;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions;
 using Microsoft.Extensions.Logging;
 using Microsoft.Net.Http.Headers;
 using NodaTime;
+using SqlStreamStore;
 
 public class DownloadFileContentRequestHandler : EndpointRequestHandler<DownloadFileContentRequest, DownloadFileContentResponse>
 {
     private readonly RoadNetworkExtractDownloadsBlobClient _client;
     private readonly IClock _clock;
+    private readonly IStreamStore _streamStore;
     private readonly EditorContext _context;
 
     public DownloadFileContentRequestHandler(
         CommandHandlerDispatcher dispatcher,
         EditorContext editorContext,
         RoadNetworkExtractDownloadsBlobClient client,
+        IStreamStore streamStore,
         IClock clock,
         ILogger<DownloadFileContentRequestHandler> logger) : base(dispatcher, logger)
     {
         _clock = clock ?? throw new ArgumentNullException(nameof(clock));
         _client = client ?? throw new ArgumentNullException(nameof(client));
         _context = editorContext ?? throw new ArgumentNullException(nameof(editorContext));
+        _streamStore = streamStore ?? throw new ArgumentNullException(nameof(streamStore));
     }
 
-    private async Task<int> CalculateRetryAfter(DownloadFileContentRequest request)
+    private async Task<int> CalculateRetryAfterAsync(DownloadFileContentRequest request, CancellationToken cancellationToken)
     {
-        return await _context.ExtractUploads.TookAverageProcessDuration(_clock
-                .GetCurrentInstant()
-                .Minus(Duration.FromDays(request.RetryAfterAverageWindowInDays)),
-            request.DefaultRetryAfter);
+        var projectionStateItem = await _context.ProjectionStates.SingleOrDefaultAsync(s => s.Name.Equals("roadregistry-editor-extractdownload-projectionhost"), cancellationToken);
+
+        if (projectionStateItem is not null)
+        {
+            var currentPosition = projectionStateItem.Position;
+            var lastPosition = await _streamStore.ReadHeadPosition(cancellationToken);
+
+            if (currentPosition < lastPosition)
+            {
+                var eventProcessorMetrics = await _context.EventProcessorMetrics.GetMetricsAsync("ExtractUploadEventProcessor", cancellationToken);
```

Comment:
Why `ExtractUploadEventProcessor`? Shouldn't it be `RoadNetworkEventProcessor`? Because that's the one the extract host is waiting for, and this retry system is to wait until the extract is ready.

Reply:
Actually we have 2 places where we calculate the retry, the projectionstates from extractdownload-projectionhost and extractupload-projectionhost is used, so this should have been ExtractDownloadEventProcessor